scripts/unified_state_manager.py

```
@@ -0,0 +1,632 @@
+                raise TimeoutError(f"Could not acquire lock (file in use)")
+            finally:
+                fcntl.lockf(lock_fd, fcntl.LOCK_UN)
+                os.close(lock_fd)
+        elif HAS_MSVCRT:
+            # Windows fallback
+            lock_fd = os.open(str(self.lock_file), os.O_CREAT | os.O_RDWR)
+            try:
+                msvcrt.locking(lock_fd, msvcrt.LK_NBLCK, 1)
+                yield lock_fd
+            except OSError:
+                raise TimeoutError(f"Could not acquire lock (file in use)")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)